misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/comparator/ModuleConstantComparator.java

```diff
@@ -0,0 +1,166 @@
+        DumbNodeListComparator<AnnotationNode> annotComparator = new DumbNodeListComparator<>(newAnnots, oldAnnots,
+                DiffKind.MODULE_CONST_ANNOTATION.toString());
+        annotComparator.computeDiff().ifPresent(metadataDiffs::add);
+
+        return metadataDiffs;
+    }
+
+    private List<Diff> compareQualifiers(ConstantDeclarationNode newNode, ConstantDeclarationNode oldNode) {
+        List<Diff> qualifierDiffs = new ArrayList<>();
+
+        // analyzes public qualifier changes
+        Optional<Token> newPublicQual = newNode.visibilityQualifier();
+        Optional<Token> oldPublicQual = oldNode.visibilityQualifier();
+        if (newPublicQual.isPresent() && oldPublicQual.isEmpty()) {
+            // public qualifier added
+            NodeDiffBuilder qualifierDiffBuilder = new NodeDiffImpl.Builder<Node>(newPublicQual.get(), null);
+            qualifierDiffBuilder
+                    .withVersionImpact(SemverImpact.MINOR)
+                    .withMessage("'public' qualifier is added to module constant '" + getModuleConstantName() + "'")
+                    .build()
+                    .ifPresent(qualifierDiffs::add);
+        } else if (newPublicQual.isEmpty() && oldPublicQual.isPresent()) {
+            // public qualifier removed
+            NodeDiffBuilder qualifierDiffBuilder = new NodeDiffImpl.Builder<Node>(null, oldPublicQual.get());
+            qualifierDiffBuilder
+                    .withVersionImpact(SemverImpact.MAJOR)
+                    .withMessage("'public' qualifier is removed from module constant '" + getModuleConstantName() + "'")
+                    .build()
+                    .ifPresent(qualifierDiffs::add);
+        }
+
+        return qualifierDiffs;
+    }
+
+    private List<Diff> compareType(ConstantDeclarationNode newNode, ConstantDeclarationNode oldNode) {
+        List<Diff> typeDiffs = new LinkedList<>();
+        Optional<TypeDescriptorNode> newType = newNode.typeDescriptor();
+        Optional<TypeDescriptorNode> oldType = oldNode.typeDescriptor();
+
+        if (newType.isEmpty() && oldType.isEmpty()) {
+            return typeDiffs;
+        } else if (newType.isPresent() && oldType.isEmpty()) {
+            NodeDiffBuilder diffBuilder = new NodeDiffImpl.Builder<>(newType.get(), null);
+            diffBuilder = diffBuilder.withVersionImpact(SemverImpact.AMBIGUOUS);
+            diffBuilder.withMessage("module constant type added");
+            diffBuilder.build().ifPresent(typeDiffs::add);
+        } else if (newType.isEmpty()) {
+            NodeDiffBuilder diffBuilder = new NodeDiffImpl.Builder<>(null, oldType.get());
+            diffBuilder = diffBuilder.withVersionImpact(SemverImpact.AMBIGUOUS);
+            diffBuilder.withMessage("module constant type removed");
+            diffBuilder.build().ifPresent(typeDiffs::add);
+        } else if (!newType.get().toSourceCode().trim().equals(oldType.get().toSourceCode().trim())) {
```

Comment:
```suggestion
        } else if (newType.kind() != oldType.kind()) {
```
Can't we use SyntaxKind for checking the type of the node?

Reply:
@lnash94 we cant :) because in here what we need to compare is the type itself.

Reply:
yeah sure, then we have to do the way with compare sourcecode 🙂